lib/class-wp-theme-json-gutenberg.php

```diff
@@ -78,13 +78,18 @@ class WP_Theme_JSON_Gutenberg {
 	);
 
 	const VALID_SETTINGS = array(
-		'border'     => array(
+		'core/navigation' => array(
+			'hasSubmenuIndicatorSetting' => null,
+			'hasItemJustificationControls' => null,
+			'hasColorSettings' => null,
```

Comment:
There is a `color` array just below, I was wondering if it should be possible to override the color settings using that instead of a new setting?

Reply:
This makes me wonder if the `hasColorSettings` prop was wrong all along, and maybe the navigation block should instead use `useSetting` to determine whether the color settings panel is rendered.

Reply:
It's probably ok for us to use `useSetting` for that one in this case as it's a general UI config item which has now been made part of Global Styles.

Reply:
In addition to what Dan mentioned about `hasColorSettings` I wonder whether `hasItemJustificationControls` can be done the same way, see discussion about that at https://github.com/WordPress/gutenberg/issues/34317 (the issue is unowned, feel free to take it).

Reply:
I've just run into another block that also uses the justification and I think this can be converted to a block supports: https://github.com/WordPress/gutenberg/issues/34872